### PR TITLE
Hp user favorites

### PR DESCRIPTION
**Pull request by segadreamgirl**
# Description
Added favorites button to everything that can be favorited. Did some housekeeping.

# Steps to test
1. `git fetch origin hp-user-favorites`
2. `git checkout hp-user-favorites`
3. You shouldn't need to restart your client, but do it if you do.
4. Make sure you're logged in and navigate to http://localhost:3000/home
5. Click on the "Add Photo to Favorites" button and make sure the text changes to "Remove photo from favorites"
6. Click on the remove button and make sure it changes back.  Click it again and go to My Hub and see if it is in the favorites carousel.
7.  Go back to home and click on a park from the list. 
8. Add it to your favorites and make sure the button changes. 
9. Remove it and add it again.
10. Go to My Hub and see if it was added to the parks block.
11. Go to blogs. Click the button on a blog to add it and then remove it. Make sure the text is changing every time, and that you leave it added to favorites before looking at your hub to see if it's there.
12. Go to the events calendar. Click on an event to see its details. Test the button's changing functionality and then see if it's in your hub once added.

### CODE REVIEW

File: src/components/event/Event.js [hunk 1]
```diff
@@ -23,18 +23,19 @@ export const Event = ({ event, setViewingEvent, setShowingEventForm, deleteEvent
       <Modal onClose={() => setViewingEvent(null)} title={`${event.name} (${event.park.name})`} className="eventModal">
         <p>From <b>{event.start_date}</b> to <b>{event.end_date}</b></p>
         <p>{event.description}</p>
+        <FavoriteBtn resource={"events"} resource_id={Number(event.id)}/>
         <>
         { isStaff? <>
-        <button href="javascript:;" onClick={() => {
+        <button className="regButton" href="javascript:;" onClick={() => {
```
Comment (stevebrownlee) [inline src/components/event/Event.js:29]: `href` is not a valid attribute on a `<button>` element.

File: src/components/favorites/favoriteBtn.js [hunk 2]
```diff
@@ -1,25 +1,234 @@
...
+          method: "DELETE",
+          body: JSON.stringify(copy)
+      })
+  }
+
+
+    //this is the section where the [resource]Favorite functions live. They render the special buttons for each resource you can favorite
+    const blogFavorite = () => {
+          let arrayOfIds = []
+          favBlogs.map((fav)=>{
```
Comment (stevebrownlee) [inline src/components/favorites/favoriteBtn.js:172]: These two lines of code should be combined....

```
let arrayOfIds = favBlogs.map(fav => fav.post.id)
```

**Review by stevebrownlee — CHANGES_REQUESTED** [review]
@segadreamgirl A few things to consider

File: src/components/event/Event.js (hunk 1 above)
Comment (segadreamgirl) [reply src/components/event/Event.js:29]: taken off!

**Comment by vanessaspear** [comment]
My only other feedback is to add a comment above the http://localhost:8000/favorites/32 fetch urls explaining the need for the "/32" in the favoriteBtn.js file.  This is more of a thought exercise, but it would be interesting to try to create a single, generic POST and DELETE function where you can pass the resource and id arguments to add/remove a favorite in the favoriteBtn.js file.  There's a lot of repeated code for the four different resources.

Awesome job on this feature!

**Review by vanessaspear — APPROVED** [review]
Nice those updates fixed it - approved!

**Pull request merged**